lib/spreewald/time_steps.rb

```diff
@@ -0,0 +1,98 @@
+# coding: UTF-8
+
+
+# Steps to travel through time using [Timecop](https://github.com/jtrupiano/timecop).
+#
+# See [this article](https://makandracards.com/makandra/1222-useful-cucumber-steps-to-travel-through-time-with-timecop) for details.
+# FILE_COMMENT_END
+
+active_support_time_available = defined?(ActiveSupport) && ([ActiveSupport::VERSION::MAJOR, ActiveSupport::VERSION::MINOR] <=> [4, 1]) != -1
+
+
+if defined?(Timecop) || active_support_time_available
+
+  module TimeHelpers
+
+    # When you have to make your rails app time zone aware you have to go 100%
+    # otherwise you are better off ignoring time zones at all.
+    # https://makandracards.com/makandra/8723-guide-to-localizing-a-rails-application
+
+    def use_timezones?
+      active_record_loaded = defined?(ActiveRecord::Base)
+      (!active_record_loaded || ActiveRecord::Base.default_timezone != :local) && Time.zone
+    end
+
+    def parse_time(str)
+      if use_timezones?
+        Time.zone.parse(str)
+      else
+        Time.parse(str)
+      end
+    end
+
+    def current_time
+      if use_timezones?
+        Time.current
+      else
+        Time.now
+      end
+    end
+
+    if defined?(Timecop)
+      def travel(duration)
+        Timecop.travel(current_time + duration)
+      end
+
+      def travel_to(point)
+        Timecop.travel(point)
+      end
+
+      def travel_back
+        Timecop.return
+      end
+    else
+      require 'active_support/testing/time_helpers'
+      include ActiveSupport::Testing::TimeHelpers
+    end
+
+  end
+
+  World(TimeHelpers)
+
+  # Example:
+  #
+  #     Given the date is 2012-02-10
+  #     Given the time is 2012-02-10 13:40
+  When /^the (?:date|time) is "?(\d{4}-\d{2}-\d{2}(?: \d{1,2}:\d{2})?)"?$/ do |time|
+    travel_to parse_time(time)
+  end.overridable
+
+  # Example:
+  #
+  #     Given the time is 13:40
+  When /^the time is "?(\d{1,2}:\d{2})"?$/ do |time_without_date|
+    travel_to parse_time(time_without_date) # date will be today
+  end.overridable
+
+  # Example:
+  #
+  #     When it is 10 minutes later
+  #     When it is a few hours earlier
+  When /^it is (\d+|a|some|a few) (seconds?|minutes?|hours?|days?|weeks?|months?|years?) (later|earlier)$/ do |amount, unit, direction|
+    amount = case amount
+      when 'a'
```

Comment:
We don't add an identation for `when` statements. Please remove the whitespace like this
```ruby
case amount
when x
  'foo'
when y
  'bar'
end
```